hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaUpdateTestWithUseJdbcMetadataDeaultsTest.java

```diff
@@ -0,0 +1,122 @@
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.MySQLDialect;
+import org.hibernate.dialect.SQLServerDialect;
+import org.hibernate.dialect.SybaseDialect;
+import org.hibernate.engine.jdbc.env.spi.IdentifierHelper;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+import org.hibernate.tool.hbm2ddl.SchemaExport;
+import org.hibernate.tool.hbm2ddl.SchemaUpdate;
+import org.hibernate.tool.hbm2ddl.SchemaValidator;
+import org.hibernate.tool.schema.JdbcMetadaAccessStrategy;
+import org.hibernate.tool.schema.TargetType;
+
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.SkipLog;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Andrea Boriero
+ */
+@RunWith(Parameterized.class)
+public class SchemaUpdateTestWithUseJdbcMetadataDeaultsTest {
+
+	@Parameterized.Parameters
+	public static Collection<String> parameters() {
+		return Arrays.asList(
+				new String[] { JdbcMetadaAccessStrategy.GROUPED.toString(), JdbcMetadaAccessStrategy.INDIVIDUALLY.toString()}
+		);
+	}
+
+	@Parameterized.Parameter
+	public String jdbcMetadataExtractorStrategy;
+
+	private File output;
+	private StandardServiceRegistry ssr;
+	private MetadataImplementor metadata;
+
+	@Before
+	public void setUp() throws IOException {
+		output = File.createTempFile( "update_script", ".sql" );
+		output.deleteOnExit();
+		ssr = new StandardServiceRegistryBuilder()
+				.applySetting( "hibernate.temp.use_jdbc_metadata_defaults", "false" )
+				.applySetting( AvailableSettings.KEYWORD_AUTO_QUOTING_ENABLED, "true" )
+				.applySetting( AvailableSettings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, jdbcMetadataExtractorStrategy )
+				.build();
+
+		final MetadataSources metadataSources = new MetadataSources( ssr );
+		metadataSources.addAnnotatedClass( TestEntity.class );
+
+		metadata = (MetadataImplementor) metadataSources.buildMetadata();
+		metadata.validate();
+	}
+
+	@After
+	public void tearsDown() {
+		new SchemaExport().setHaltOnError( true )
+				.setOutputFile( output.getAbsolutePath() )
+				.setFormat( false )
+				.drop( EnumSet.of( TargetType.DATABASE ), metadata );
+		StandardServiceRegistryBuilder.destroy( ssr );
+	}
+
+	@Test
+	public void testSchemaUpdateWorksWhenUseJdbcMetadataDefaultsIsFalse() throws Exception {
+
+		new SchemaUpdate().setHaltOnError( true )
+				.execute( EnumSet.of( TargetType.DATABASE ), metadata );
+
+		new SchemaValidator().validate( metadata );
+
+		new SchemaUpdate().setHaltOnError( true )
+				.setOutputFile( output.getAbsolutePath() )
+				.setFormat( false )
+				.execute( EnumSet.of( TargetType.DATABASE, TargetType.SCRIPT ), metadata );
+
+		final String fileContent = new String( Files.readAllBytes( output.toPath() ) );
```

Comment:
should you use `getAbsolutePath` here as well?

Reply:
`getAbsolutePath()` returns a `String` while `Files.readAllBytes` requires a `Path`